### PR TITLE
MES-2170 - Test Report Validator Provider

### PR DESCRIPTION
**Pull request by Cavalier07**
## Description and relevant Jira numbers

- Adds Validator to check legal requirements have been met for a test
- Fixes one of the labels for Kaisa

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [x] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [x] PR link added to JIRA
- [ ] Reviewers selected in Github
- [x] Any new reusable component/widget added to component library on Confluence

- [x] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

### CODE REVIEW

File: src/modules/tests/test-data/test-data.selector.ts [hunk 1]
```diff
@@ -102,3 +106,32 @@ export const getSelectedTellMeQuestionText = (state: VehicleChecks) => {
...
+      tellMeQuestionOutcome === CompetencyOutcome.P ||
+      tellMeQuestionOutcome === CompetencyOutcome.DF ||
+      tellMeQuestionOutcome === CompetencyOutcome.S ||
+      tellMeQuestionOutcome === CompetencyOutcome.D
+  );
+};
+
+export const getCatBLegalRequirements = (data: TestData): CatBLegalRequirements => {
+  return {
+    normalStop1: data.testRequirements.normalStart1 || false,
```
Comment (rossfauldsbjss) [inline src/modules/tests/test-data/test-data.selector.ts]: confusing - normalstop1 and 2 are based on the result of normalStart1 and 2?
why the naming difference?
probably nothing wrong with this - may be my lack of familiarity with these competencies.

File: src/pages/test-report/__tests__/test-report.selector.spec.ts [hunk 2]
```diff
@@ -18,4 +19,9 @@ describe('TestReportSelectors', () => {
       expect(isDangerousMode(state)).toBeTruthy();
     });
   });
+  describe('isTestValid', () => {
+    it('should return if the test is valid', () => {
```
Comment (rossfauldsbjss) [inline src/pages/test-report/__tests__/test-report.selector.spec.ts]: it text should be - should return true if the test is valid

Comment (Cavalier07) [reply src/pages/test-report/__tests__/test-report.selector.spec.ts]: Fixed

File: src/modules/tests/test-data/test-data.selector.ts (hunk 1 above)
Comment (Cavalier07) [reply src/modules/tests/test-data/test-data.selector.ts]: Fixed - Nice Spot

**Pull request merged**